Source/MQTTnet.Extensions.ManagedClient/ManagedMqttClient.cs

```diff
@@ -193,8 +193,9 @@ public Task UnsubscribeAsync(IEnumerable<string> topics)
 
         public void Dispose()
         {
-            _connectionCancellationToken?.Dispose();
-            _publishingCancellationToken?.Dispose();
+            _mqttClient?.Dispose();
```

Comment:
I am wondering why this is done before the Stop calls. I assume Stop should happen first. Otherwise it may happen that an already disposed object is used.

Reply:
Yes this was very uncommon but it was required because if I cancel the _connectionCancellationToken first the MaintainConnectionAsync’s finnaly block called _mqttClient.DisconnectAsync and therefore did a regular disconnect.